tests/components/select/test_device_condition.py

```diff
@@ -0,0 +1,185 @@
+    assert_lists_same(conditions, expected_conditions)
+
+
+async def test_if_selected_option(
+    hass: HomeAssistant, calls: list[ServiceCall]
+) -> None:
+    """Test for selected_option conditions."""
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {"platform": "event", "event_type": "test_event1"},
+                    "condition": [
+                        {
+                            "condition": "device",
+                            "domain": DOMAIN,
+                            "device_id": "",
+                            "entity_id": "select.entity",
+                            "type": "selected_option",
+                            "option": "option1",
+                        }
+                    ],
+                    "action": {
+                        "service": "test.automation",
+                        "data": {
+                            "result": "option1 - {{ trigger.platform }} - {{ trigger.event.event_type }}"
+                        },
+                    },
+                },
+                {
+                    "trigger": {"platform": "event", "event_type": "test_event2"},
+                    "condition": [
+                        {
+                            "condition": "device",
+                            "domain": DOMAIN,
+                            "device_id": "",
+                            "entity_id": "select.entity",
+                            "type": "selected_option",
+                            "option": "option2",
+                        }
+                    ],
+                    "action": {
+                        "service": "test.automation",
+                        "data": {
+                            "result": "option2 - {{ trigger.platform }} - {{ trigger.event.event_type }}"
+                        },
+                    },
+                },
+            ]
+        },
+    )
+
+    # Test with non existing entity
+    hass.bus.async_fire("test_event1")
+    hass.bus.async_fire("test_event2")
+    await hass.async_block_till_done()
+    assert len(calls) == 0
+
+    hass.states.async_set(
+        "select.entity", "option1", {"options": ["option1", "option2"]}
+    )
+    hass.bus.async_fire("test_event1")
+    hass.bus.async_fire("test_event2")
+    await hass.async_block_till_done()
+    assert len(calls) == 1
+    assert calls[0].data["result"] == "option1 - event - test_event1"
+
+    hass.states.async_set(
+        "select.entity", "option2", {"options": ["option1", "option2"]}
+    )
+    hass.bus.async_fire("test_event1")
+    hass.bus.async_fire("test_event2")
+    await hass.async_block_till_done()
+    assert len(calls) == 2
+    assert calls[1].data["result"] == "option2 - event - test_event2"
+
+
+async def test_get_condition_capabilities(hass: HomeAssistant) -> None:
```

Comment:
Parametrize this test to make sure it works both when there is a state and when the options are fetched from the entity registry,